src/components/image/index.web.tsx

```diff
@@ -0,0 +1,236 @@
+    if (nextProps.source !== prevState.prevSource) {
+      return {
+        error: false,
+        prevSource: nextProps.source
+      };
+    }
+    return null;
+  }
+
+  isGif() {
+    if (Constants.isAndroid) {
+      const {source} = this.props;
+      const url = _.get(source, 'uri');
+      const isGif = /(http(s?):)([/|.|\w|\s|-])*\.gif/.test(url ?? '');
+      return isGif;
+    }
+  }
+
+  shouldUseImageBackground() {
+    const {overlayType, customOverlayContent} = this.props;
+
+    return !!overlayType || this.isGif() || !_.isUndefined(customOverlayContent);
+  }
+
+  getVerifiedSource(source?: ImagePropTypes.source) {
+    if (_.get(source, 'uri') === null || _.get(source, 'uri') === '') {
+      // @ts-ignore
+      return {...source, uri: undefined};
+    }
+    return source;
+  }
+
+  getImageSource() {
+    const {assetName, assetGroup, source} = this.props;
+
+    if (!_.isUndefined(assetName)) {
+      return getAsset(assetName, assetGroup);
+    }
+    if (this.sourceTransformer) {
+      return this.sourceTransformer(this.props);
+    }
+
+    return this.getVerifiedSource(source);
+  }
+
+  onError = (event: NativeSyntheticEvent<ImageErrorEventData>) => {
+    if (event.nativeEvent.error) {
+      this.setState({error: true});
+      this.props.onError?.(event);
+    }
+  };
+
+  renderSvg = () => {
+    const {source, recorderTag, ...others} = this.props;
+    return <SvgImage data={source} fsTagName={recorderTag} {...others}/>;
+  };
+
+  renderImageWithContainer = () => {
+    const {style, cover, modifiers, width, height} = this.props;
+    const {margins} = modifiers;
+
+    return (
+      <View style={[{width, height}, margins, style, styles.errorImageContainer, cover && styles.coverImage]}>
+        {this.renderImage(true)}
+      </View>
+    );
+  };
+
+  renderImage = (useImageInsideContainer: boolean) => {
+    const {error} = this.state;
+    const source = error ? this.getVerifiedSource(this.props.errorSource) : this.getImageSource();
+    const {
+      tintColor,
+      style,
+      width,
+      height,
+      supportRTL,
+      cover,
+      aspectRatio,
+      overlayType,
+      overlayIntensity,
+      overlayColor,
+      customOverlayContent,
+      modifiers,
+      ...others
+    } = this.props;
+    const shouldFlipRTL = supportRTL && Constants.isRTL;
+    const {margins} = modifiers;
+    const imageViewStyle = [
+      tintColor && {tintColor},
+      shouldFlipRTL && styles.rtlFlipped,
+      width && {width},
+      height && {height},
+      cover && styles.coverImage,
+      this.isGif() && styles.gifImage,
+      aspectRatio && {aspectRatio},
+      !useImageInsideContainer && margins,
+      useImageInsideContainer && styles.containImage,
+      style,
+      useImageInsideContainer && styles.shrink
+    ];
+
+    const finalSource = source.uri ?? source;
+
+    return (
+      // @ts-expect-error
+      <View style={imageViewStyle}>
```

Comment:
Why do you add this extra `View` and not pass the style to the `img` component?

Reply:
@OrenZak is right